### PR TITLE
Remove deprecations for main/gz-sim8

### PR DESCRIPTION
**Pull request by mjcarroll**
# 🎉 New feature

## Summary

Remove `ignition` headers and things that have been marked as deprecated in gz-sim8

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -263,13 +263,3 @@ endif()
 if(TARGET doc)
   file(COPY ${CMAKE_SOURCE_DIR}/tutorials/files/ DESTINATION ${CMAKE_BINARY_DIR}/doxygen/html/files/)
 endif()
-
-# TICKTOCK MASTER HEADER (to bypass GZ_DESIGNATION)
-# TODO(CH3): Deprecated. Remove on tock
-configure_file(
```
Comment (azeey) [inline CMakeLists.txt:269]: No idea what this was actually doing. AFAIK, `GZ_DESIGNATION` will be `sim`, so this is a noop?

File: src/SystemLoader.cc [hunk 2]
```diff
@@ -65,37 +65,18 @@ class gz::sim::SystemLoaderPrivate
...
-      else
-      {
-        gzwarn << "Found plugin [" << filename
-               << "] using deprecated environment variable ["
-               << pluginPathEnvDeprecated << "]. Please use ["
-               << pluginPathEnv << "] instead." << std::endl;
+        gzerr << "Failed to load system plugin [" << filename <<
+                  "] : couldn't find shared library." << std::endl;
       }
+      return false;
```
[the remark this answers is not in the document]
Comment (mjcarroll) [reply src/SystemLoader.cc:79]: I think that the condition is that if the library isn't found, the pathToLib is empty.  If pathToLib is empty then we have historically always returned false: https://github.com/gazebosim/gz-sim/blob/gz-sim7/src/SystemLoader.cc#L98-L122

Checking that the plugin name is "gz::sim" is what we are using to determine if the error should be printed.  We don't want to print the error in the case that it is the levels feature and set to `gz::sim`.

File: src/systems/perfect_comms/PerfectComms.cc [hunk 3]
```diff
@@ -110,10 +110,3 @@ GZ_ADD_PLUGIN(PerfectComms,
                     gz::sim::System,
                     comms::ICommsModel::ISystemConfigure,
                     comms::ICommsModel::ISystemPreUpdate)
-
-GZ_ADD_PLUGIN_ALIAS(PerfectComms,
```
[the remark this answers is not in the document]
Comment (mjcarroll) [reply src/systems/perfect_comms/PerfectComms.cc:114]: 98dd43ea0

File: test/integration/wheel_slip.cc [hunk 4]
```diff
@@ -240,7 +240,7 @@ TEST_F(WheelSlipTest, GZ_UTILS_TEST_DISABLED_ON_WIN32(TireDrum))
   // const double kp = surfaceContactOde->GetElement("kp")->Get<double>();
   // ASSERT_EQ(kp, 250e3);
 
-  double modelMass = 0.0;
+  // double modelMass = 0.0;
```
[the remark this answers is not in the document]
Comment (mjcarroll) [reply test/integration/wheel_slip.cc]: ce5b12b88

File: CMakeLists.txt (hunk 1 above)
Comment (mjcarroll) [reply CMakeLists.txt:269]: Same here.  It said remove in 8, so I did.  Maybe @methylDragon has an idea?

File: src/SystemLoader.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (azeey) [reply src/SystemLoader.cc:79]: Looking further back in history, I think you're right.

**Review by azeey — APPROVED** [review]

**Comment by azeey** [comment]
codecheck is failing.

**Comment by mjcarroll** [comment]
> codecheck is failing.

arg dd25b33d3

**Review by ahcorde — APPROVED** [review]

**Pull request merged**